server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetAliasesAction.java

```diff
@@ -84,59 +85,22 @@ static RestResponse buildRestResponse(
     ) throws Exception {
         final Set<String> indicesToDisplay = new HashSet<>();
         final Set<String> returnedAliasNames = new HashSet<>();
-        for (final Map.Entry<String, List<AliasMetadata>> cursor : responseAliasMap.entrySet()) {
-            for (final AliasMetadata aliasMetadata : cursor.getValue()) {
-                if (aliasesExplicitlyRequested) {
-                    // only display indices that have aliases
-                    indicesToDisplay.add(cursor.getKey());
-                }
-                returnedAliasNames.add(aliasMetadata.alias());
-            }
-        }
-        dataStreamAliases.entrySet()
-            .stream()
-            .flatMap(entry -> entry.getValue().stream())
-            .forEach(dataStreamAlias -> returnedAliasNames.add(dataStreamAlias.getName()));
+        if (aliasesExplicitlyRequested) {
+            responseAliasMap.entrySet().stream().filter(entry -> entry.getValue().isEmpty() == false).forEach(entry -> {
```

Comment:
Streams are themselves pretty expensive in terms of both allocations and general CPU usage. Could we just use a plain old-fashioned loop here?